module/Photo/src/Photo/Mapper/Album.php

```diff
@@ -151,6 +151,61 @@ public function getRootAlbums()
     }
 
     /**
+     * Gets all root albums containing photos made between the specified dates
+     *
+     * @param $start \DateTime start date and time
+     * @param $end \DateTime end date and time
+     *
+     * @return array of \Photo\Model\Album
+     */
+    public function getAlbumsInDateRange($start, $end) {
+        $qb = $this->em->createQueryBuilder();
+
+        $qb->select('a')
+            ->from('Photo\Model\Album', 'a')
+            ->where('a.parent IS NULL')
+            ->andWhere('a.startDateTime BETWEEN ?1 AND ?2')
+            ->setParameter(1, $start)
+            ->setParameter(2, $end);
+
+        return $qb->getQuery()->getResult();
+    }
+
+    /**
+     * Returns the root album containing the most recent photos
+     *
+     * @return \Photo\Model\Album
+     */
+    public function getLatestAlbum() {
+        $qb = $this->em->createQueryBuilder();
+
+        $qb->select('a')
+            ->from('Photo\Model\Album', 'a')
+            ->where('a.parent IS NULL')
+            ->setMaxResults(1)
+            ->orderBy('a.startDateTime', 'DESC');
+
+        //TODO: this is going to break
+        return $qb->getQuery()->getResult()[0];
+    }
+
+    /**
+     * Returns the root album containing the oldest photos
+     *
+     * @return \Photo\Model\Album
+     */
+    public function getOldestAlbum() {
+        $qb = $this->em->createQueryBuilder();
+
+        $qb->select('a')
+            ->from('Photo\Model\Album', 'a')
+            ->where('a.parent IS NULL')
+            ->setMaxResults(1)
+            ->orderBy('a.startDateTime', 'ASC');
+
+        return $qb->getQuery()->getResult()[0];
```

Comment:
Dit kan net zo goed breken als the getLatestAlbum(). Daarbij is de naamgeving insconsistent. GetOldestAlbum hoort een getNewestAlbum te hebben; getLatestAlbum() moet een getFirstAlbum() hebben. Zoals de naamgeving nu is schept het verwarring